query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/QueryRequestToPromqlConverter.java

```diff
@@ -0,0 +1,259 @@
+package org.hypertrace.core.query.service.prometheus;
+
+import static org.hypertrace.core.query.service.QueryRequestUtil.getLogicalColumnNameForSimpleColumnExpression;
+
+import com.google.common.collect.Lists;
+import com.google.protobuf.ByteString;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.commons.codec.binary.Hex;
+import org.hypertrace.core.query.service.ExecutionContext;
+import org.hypertrace.core.query.service.QueryRequestUtil;
+import org.hypertrace.core.query.service.QueryTimeRange;
+import org.hypertrace.core.query.service.api.Expression;
+import org.hypertrace.core.query.service.api.Expression.ValueCase;
+import org.hypertrace.core.query.service.api.Filter;
+import org.hypertrace.core.query.service.api.LiteralConstant;
+import org.hypertrace.core.query.service.api.Operator;
+import org.hypertrace.core.query.service.api.QueryRequest;
+import org.hypertrace.core.query.service.api.Value;
+import org.hypertrace.core.query.service.prometheus.PrometheusViewDefinition.MetricConfig;
+
+class QueryRequestToPromqlConverter {
```

Comment:
nit: before this file becomes the monstrosity that is the pinot query converter, it may make sense to break down various conversions into separate files.

Reply:
Moved filter converter out